numpy/core/records.py

```diff
@@ -610,8 +613,6 @@ def fromarrays(arrayList, dtype=None, shape=None, formats=None,
         # and determine the formats.
         formats = []
         for obj in arrayList:
-            if not isinstance(obj, ndarray):
-                raise ValueError("item in the array list must be an ndarray.")
             formats.append(obj.dtype.str)
```

Comment:
I wonder if you could get a greater gain by just appending `obj.dtype` here.

Reply:
I will try this out

Reply:
@eric-wieser 
should we keep this change?